Tests/GeoLib/TestPlanePlaneIntersection.cpp

```diff
@@ -0,0 +1,235 @@
+/**
+ * @date 2015-10-19
+ *
+ * @copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <limits>
+
+#include <gtest/gtest.h>
+#include <autocheck/autocheck.hpp>
+
+#include "Tests/MathLib/AutoCheckTools.h"
+
+#include "MathLib/MathTools.h"
+
+#include "GeoLib/AnalyticalGeometry.h"
+#include "GeoLib/Plane.h"
+
+namespace ac = autocheck;
+
+struct GeoLibComputePlanePlaneIntersection : public ::testing::Test
+{
+	// to generate arbitrary straight line segments
+	ac::randomTupleGenerator<double, 3> tuple_generator;
+	ac::cons_generator<MathLib::Point3d, ac::randomTupleGenerator<double, 3>>
+	    points_gen{tuple_generator};
+
+	ac::gtest_reporter gtest_reporter;
+};
+
+TEST_F(GeoLibComputePlanePlaneIntersection, TestPlanePlaneIntersection)
+{
+	auto checkPlanePlaneIntersection =
```

Comment:
Could you add a comment like: "Takes a list of four points (p0, ... p3) , constructs two planes (p0, p1, p2), (p1, p2, p3), computes their intersection".
Sure, one can reaad that information also directly from the code, but a comment would save one minute of thinking.